runtime/amd64.S

```diff
@@ -232,7 +232,12 @@
 /* Stack switching operations */
 /******************************************************************************/
 
-/* Switch from OCaml to C stack. Clobbers %r10, %r11. */
+/* Switch from OCaml to C stack. Clobbers %r10, %r11.
+
+   If a C function is called which might call back into OCaml,
+   then nothing may be pushed to the C stack between SWITCH_OCAML_TO_C
```

Comment:
I don't understand this comment. In the new code (caml_c_call_copy_stack_args), new things are exactly "pushed to the C stack between SWITCH_OCAML_TO_C and the next C call". Do you mean the OCaml stack?
Same remark applies to the matching comment in arm64.S.

Reply:
> In the new code (caml_c_call_copy_stack_args), new things are exactly "pushed to the C stack between SWITCH_OCAML_TO_C and the next C call".

No, they aren't. The next C call after `SWITCH_OCAML_TO_C` is in fact the call to `caml_c_call_copy_stack_args`, which happens before any pushing. This is why that function is an out-of-line call, to ensure that the last frame pointer that points into the OCaml stack is at a constant offset from the stack position at `SWITCH_OCAML_TO_C` time.